### ISSUE TITLE
Can't list dirty packages: diffing directory for dirty packages, exit status 128  script returned exit code 1

### ISSUE THREAD

**Issue by shojaeix** [digitalocean/gta#45]
Hi, 

I'm receiving an error when running GTA on my Jenkins pipeline

First I run the following command successfully: 
`gta -include {repo_address}/ -base origin/develop | sed 's/{repo_address}\///g' | sed 's/tools\/[^[:space:]]\+//g'`

but then I run the following command and it fails:
 
My command: `gta -include {repository_address}/ -base origin/develop`

The response: `main.go:82: can't list dirty packages: diffing directory for dirty packages, exit status 128 script returned exit code 1`

 
 Can someone help me here? couldn't find anything on the web. 
 Thank you

**Comment by bhcleek** [comment]
Here are some questions whose answers may help us get to the root of the problem you're experiencing:
* For the command that's running successfully, is that being run in a bash script?
* Is `set -o pipefail` set?Is `gta` being run within the directory that contains your main module? 
* Is `git` installed?
* Is `git` in `$PATH`?

**Comment by brunoluiz** [comment]
I came across similar issues and I realised that actually I needed to vendor my repository to make it work. Perhaps it is the same for @shojaeix?

- https://github.com/digitalocean/gta/issues/27#issuecomment-928047840
- https://github.com/digitalocean/gta/issues/15#issuecomment-772634797

(Is not clear from the documentation though)

**Comment by matthewd98** [comment]
@brunoluiz you can avoid vendoring your dependencies if you want. I posted a [solution](https://github.com/digitalocean/gta/issues/27#issuecomment-2349652478) in related issue #27

**Comment by alecthomas** [comment]
This is still present now FWIW:

```
🐚 ~/dev/ftl $ go version
go version go1.23.4 darwin/arm64
🐚 ~/dev/ftl $ go install github.com/digitalocean/gta/cmd/gta@latest
🐚 ~/dev/ftl $ gta -buildable-only -include "$(go list -f '{{ .Module.Path }}')/"
10:16:49 main.go:82: can't list dirty packages: diffing directory for dirty packages, exit status 128
```

**Comment by matthewd98** [comment]
> This is still present now FWIW:
> 
> ```
> 🐚 ~/dev/ftl $ go version
> go version go1.23.4 darwin/arm64
> 🐚 ~/dev/ftl $ go install github.com/digitalocean/gta/cmd/gta@latest
> 🐚 ~/dev/ftl $ gta -buildable-only -include "$(go list -f '{{ .Module.Path }}')/"
> 10:16:49 main.go:82: can't list dirty packages: diffing directory for dirty packages, exit status 128
> ```

Yes, that's expected based on the previous comments. The tool requires the dependencies to be vendored (by running go mod vendor in your repository). This is how it identifies which packages are affected by a dependency upgrade.

**Comment by matthewd98** [comment]
> This is still present now FWIW:
> 
> ```
> 🐚 ~/dev/ftl $ go version
> go version go1.23.4 darwin/arm64
> 🐚 ~/dev/ftl $ go install github.com/digitalocean/gta/cmd/gta@latest
> 🐚 ~/dev/ftl $ gta -buildable-only -include "$(go list -f '{{ .Module.Path }}')/"
> 10:16:49 main.go:82: can't list dirty packages: diffing directory for dirty packages, exit status 128
> ```

Yes, that's expected based on the previous comments. The tool needs the dependencies to be vendored (by running go mod vendor in your repository) to be able to determine which packages are impacted by a dependency upgrade.

**Comment by alecthomas** [comment]
Yeah I understand that, but vendoring is so uncommon that this still seems like a bug to me.

**Comment by ndombroski** [comment]
It appears the cause of this issue has been identified, and the lack of a descriptive error message in this case has been addressed by https://github.com/digitalocean/gta/pull/56.
